### DISCUSSION TITLE
Pause/Resume Record to File Feature

### DISCUSSION THREAD

**Discussion by kenle** [HaishinKit/HaishinKit.swift#1554] [Idea]
Hi @shogo4405 ,

Would it be simple to add in a pause/resume record to file feature?

Two methods of possibly accomplishing this is to not append frames to the file when in the pause state, but will need to update time stamps appropriately, the other method is to create intermediate files and merge them at the end when stopping record.

Thanks,
- Ken

**Comment by shogo4405** [comment]
I have no plans to implement it, as I don’t see a need for the feature.

**Comment by kenle** [comment]
Oh ok.  Users who livestream and record to file for long videos have portions they want to pause and not record.  I know there is blackout for livestream but for file record users don't have an option to pause at the moment.  I had users asking about this.  I'll try to implement in a fork in the future.  Thanks.

**Reply by shogo4405** [reply]
I thought it wasn't necessary because it is synced with stream.isPaused, but isn't the recording reflecting when stream.isPaused?

**Reply by kenle** [reply]
I tested it in version 1.7.6 and it blacked out the stream but the file continued to record like normal.

Upon observing the Recorder class I noticed that it is independent of the isPaused flag.  The Recorder class has an 'isRunning' flag that is only set by 'startRunning' and 'stopRunning' functions and is checked by the 'append' function.  I tried adding a 'pauseRunning' and 'resumeRunning' functions that updated the 'isRunning' flag without closing the file, what resulted is the video will freeze the video frames during the paused portions.  In order to omit the paused portions more work needs to be done to offset the time stamps when appending the video frames upon resume.   The other option is to record separate files and merge them.

I observed version 1.9.3 and 2.0.0-beta and the logic appears to be almost similar if I'm not mistaken.
